src/Tribe/Aggregator/Processes/Import_Events.php

```diff
@@ -0,0 +1,365 @@
+			}
+		}
+	}
+
+	/**
+	 * Returns the `meta_key` that will be used to store the transitional data
+	 * in linked post for this import process.
+	 *
+	 * @since TBD
+	 *
+	 * @param null $transitional_id
+	 *
+	 * @return string
+	 */
+	public function get_transitional_meta_key( $transitional_id = null ) {
+		if ( null === $transitional_id ) {
+			$transitional_id = $this->transitional_id;
+		}
+
+		return '_tribe_import_' . $transitional_id;
+	}
+
+	/**
+	 * Sets the final part `meta_key` that should be used to store transitional
+	 * information for this import process.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $transitional_id
+	 */
+	public function set_transitional_id( $transitional_id ) {
+		$this->transitional_id = $transitional_id;
+	}
+
+	/**
+	 * Overrides the parent `save` method to save some additonal data.
+	 *
+	 * @since TBD
+	 *
+	 * @return Tribe__Events__Aggregator__Processes__Import_Events
+	 */
+	public function save() {
+		add_filter( "tribe_process_queue_{$this->identifier}_save_data", array( $this, 'save_data' ) );
+
+		return parent::save();
+	}
+
+	/**
+	 * Overrides the parent `update` method to save some additonal data.
+	 *
+	 * @since TBD
+	 *
+	 * @return Tribe__Events__Aggregator__Processes__Import_Events
+	 */
+	public function update( $key, $data ) {
+		add_filter( "tribe_process_queue_{$this->identifier}_update_data", array( $this, 'save_data' ) );
+
+		return parent::update( $key, $data );
+	}
+
+	/**
+	 * Saves some additional data on the record to keep track of the progress.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $save_data
+	 *
+	 * @return array
+	 */
+	public function save_data( array $save_data = array() ) {
+		$save_data['record_id'] = $this->record_id;
+
+		return $save_data;
+	}
+
+	/**
+	 * Returns this import process record post ID.
+	 *
+	 * @since TBD
+	 *
+	 * @return int
+	 */
+	public function get_record_id() {
+		return $this->record_id;
+	}
+
+	/**
+	 * Sets this import process record ID.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $record_id
+	 */
+	public function set_record_id( $record_id ) {
+		$this->record_id = $record_id;
+	}
+
+	/**
+	 * Handles the real import.
+	 *
+	 * In short: if an event has dependencies and those are not yet all in place then the event
+	 * will be re-queued; otherwise it's inserted.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $item
+	 *
+	 * @return array|false Either the event data to requeue or `false` if done.
+	 */
+	protected function task( $item ) {
+		$record_id             = $this->record_id = $item['record_id'];
+		$data                  = (array) $item['data'];
+		$this->transitional_id = filter_var( $item['transitional_id'], FILTER_SANITIZE_STRING );
+
+		$dependencies = $this->parse_linked_post_dependencies( $data );
+
+		if ( empty( $dependencies ) ) {
+			$this->has_dependencies = false;
+			$activity               = $this->insert_event( $record_id, (object) $data );
```

Comment:
Doesn't `insert_event()` expect an array rather than an object (or perhaps we need to update the docblock for that method)? Same question applies to line 188.

Reply:
The doc-block needed to be updated, it expects an object, in fact.